pkg/controllers/provisioning/scheduling/nodeclaim.go

```diff
@@ -284,8 +284,8 @@ func fits(instanceType *cloudprovider.InstanceType, requests v1.ResourceList) bo
 
 func hasOffering(instanceType *cloudprovider.InstanceType, requirements scheduling.Requirements) bool {
```

Comment:
IMO this should also be made generic, or maybe we can leverage `offerings.Compatible` directly and then further filter on available.